magicblock-processor/tests/scheduling.rs

```diff
@@ -0,0 +1,788 @@
+
+    // The final value *must* be the last value we wrote (19)
+    assert_eq!(
+        final_value,
+        (num_txs - 1) as u8,
+        "Final account data should be the last value written"
+    );
+}
+
+/// **Scenario 7: Serial Transfer Chain (Asserts Order)**
+///
+/// Schedules N transactions in a dependency chain (A->B, B->C, C->D).
+/// Asserts that they are executed sequentially in the *exact order*
+/// they were scheduled, as each tx depends on the lock from the previous.
+async fn scenario_serial_transfer_chain(executors: u32) {
+    let mut env = setup_env(executors);
+    let num_txs = 20;
+    let num_accounts = num_txs + 1;
+    let accounts = create_accounts(&env, num_accounts);
+    let mut initial_balances = HashMap::new();
+    let mut txs = vec![];
+
+    for acc in &accounts {
+        initial_balances.insert(acc, env.get_account(*acc).lamports());
+    }
+
+    for i in 0..num_txs {
+        let from = &accounts[i];
+        let to = &accounts[i + 1];
+        let (tx, _) = build_transfer_tx(&env, from, to, TRANSFER_AMOUNT);
+        txs.push(tx);
+        // we force a different slot to generate a new signature
+        env.advance_slot();
+    }
+
+    // Schedule sequentially and get the ordered signatures
+    let sigs_order = schedule_all_and_get_order(&env, txs).await;
+    env.run_scheduler();
+    env.advance_slot();
+
+    // Assert that the statuses were received in the *exact* order of scheduling
+    assert_statuses_in_order(&env, sigs_order, DEFAULT_TIMEOUT).await;
+
+    // Verify final state
+    env.advance_slot();
+    // First account (A) should have lost lamports
+    let first_acc = &accounts[0];
+    assert_eq!(
+        env.get_account(*first_acc).lamports(),
+        initial_balances.get(first_acc).unwrap() - TRANSFER_AMOUNT
+    );
+
+    // Intermediary accounts (B, C, ...) should have net-zero change
+    for int_acc in accounts.iter().take(num_txs).skip(1) {
+        assert_eq!(
+            env.get_account(*int_acc).lamports(),
+            *initial_balances.get(int_acc).unwrap(),
+            "Intermediary account {} lamports changed",
+            int_acc
+        );
+    }
+
+    // Last account should have gained lamports
+    let last_acc = &accounts[num_accounts - 1];
+    assert_eq!(
+        env.get_account(*last_acc).lamports(),
+        initial_balances.get(last_acc).unwrap() + TRANSFER_AMOUNT
+    );
+}
+
+// #################################################################
+// ## Test Matrix
+// #################################################################
+
+/// Macro to generate tests for different executor counts
+macro_rules! test_scenario {
+    ($scenario_fn:ident, $test_name:ident, $executors:expr) => {
+        #[tokio::test]
+        #[allow(non_snake_case)]
+        async fn $test_name() {
+            $scenario_fn($executors).await;
+        }
+    };
+}
+
+// --- Test Scenario 1: Parallel Transfers (No Conflicts) ---
+test_scenario!(
```

Comment:
That is a pain to use in IDE, I can't just launch test in debug using RustRover without special shaman dances anymore with configs. 

Practically, I'm not even sure we're saving a lot of lines here, could be just normal
```        
#[tokio::test]
async fn $test_name()
```

Please replace this

Reply:
Fixed in 908e788c5dff6a9affe56d23efa0b3a6bd0d4f43